### PR TITLE
refactored document builder factory

### PR DESCRIPTION
**Pull request by bilalyasar**
XML parser should be configured so that it does not allow external entities as part of an incoming XML document. so i disabled the doctype declaration which is enough according to my researches.

### CODE REVIEW

**Comment by mesutcelik** [comment]
@bilalyasar , can you please add description for the PR?

**Comment by bilalyasar** [comment]
added description

**Comment by jerrinot** [comment]
@bilalyasar: Is it possible to write a test for this?

**Comment by bilalyasar** [comment]
@jerrinot , added test.. thanks.

File: hazelcast-client/src/main/java/com/hazelcast/client/config/XmlClientConfigBuilder.java [hunk 1]
```diff
@@ -131,6 +131,7 @@ public XmlClientConfigBuilder() {
     protected Document parse(InputStream inputStream) throws Exception {
         DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
         dbf.setNamespaceAware(true);
+        dbf.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
```
Comment (jerrinot) [inline hazelcast-client/src/main/java/com/hazelcast/client/config/XmlClientConfigBuilder.java]: I believe this is a Xerces 2 -specific feature. Other parsers will probably ignore it. 
Anyway, we accept configuration input from trusted sources only. So I think this is ok.

**Comment by jerrinot** [comment]
:+1: once green

**Comment by bilalyasar** [comment]
it is green now.

**Comment by mesutcelik** [comment]
:+1:

**Pull request merged**